filebeat/input/filestream/environment_test.go

```diff
@@ -20,8 +20,8 @@
 package filestream
```

Comment:
Are these changes intentional here? Looks like all this refactoring is unrelated to the described change.

Reply:
There are two set of changes on this file:
1. Updates to how the `inputTestingEnvironment` handle logs:
    - Use a in-memory logger instead of the global logger
    - On failure save the logs to a file in the temporary folder for easy of debugging
2. Fixing all the lint warnings on this file/PR.

Reply:
The `#1` changes, I did while working on this PR, so I ended up keeping them because I found them helpful.

Reply:
Would it be hard to split these changes?

I'd prefer to have our PRs atomic and not addressing multiple things at once. In case we needed to revert one of these changes it would affect the other.

Reply:
Yes, I can split them into another PR.

Reply:
The split PR: https://github.com/elastic/beats/pull/44071